### ISSUE TITLE
[bug] Deleting megaport_aws_connection does not delete anything

### ISSUE THREAD

**Issue by kevinmartin** [megaport/terraform-provider-megaport#20]
I created several AWS HC connections and upon remove the configuration for them, Terraform "destroyed" the resources and removed them from state, but the AWS DX Connections still exist on the AWS account and the VXCs also exist on the Megaport portal.

Example Terraform that applied before destroy:
```hcl
resource "megaport_aws_connection" "example" {
  vxc_name   = "example"
  rate_limit = 100

  a_end {
    requested_vlan = 1234
  }

  csp_settings {
    attached_to          = data.megaport_port.example.id
    requested_product_id = data.megaport_partner_port.example.id
    requested_asn        = 65002
    amazon_asn           = 65001
    amazon_account       = 123456789123
    hosted_connection    = true
  }
}
```

Terraform logs for applying the removal:
```sh
Do you want to perform these actions in workspace "example"?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

megaport_aws_connection.example: Destroying... [id=uuid]
megaport_aws_connection.example: Still destroying... [id=uuid, 10s elapsed]
megaport_aws_connection.example: Still destroying... [id=uuid, 20s elapsed]
megaport_aws_connection.example: Still destroying... [id=uuid, 30s elapsed]
megaport_aws_connection.example: Still destroying... [id=uuid, 40s elapsed]
megaport_aws_connection.example: Destruction complete after 43s

Apply complete! Resources: 0 added, 0 changed, 1 destroyed.
```

Expected result: VXC and/or AWS connection to be deleted.
Actual result: Neither are deleted.

**Comment by kevinmartin** [comment]
Note: I tried deleting just now in the UI and I was blocked: The AWS DX connections were not deleted.

If that's the case, the resource should have failed instead of falsely destroying.

**Comment by pushee** [comment]
Hi @kevinmartin - a fix was released with v0.2.3 which passes delete failures properly to terraform which should resolve this issue. Please try it out and let us know how you find it.
